async-query-core/src/main/java/org/opensearch/sql/spark/parameter/GeneralSparkParameterComposer.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.parameter;
+
+import org.opensearch.sql.spark.asyncquery.model.AsyncQueryRequestContext;
+import org.opensearch.sql.spark.dispatcher.model.DispatchQueryRequest;
+
+/** Compose spark submit parameters. See {@link SparkParameterComposerCollection}. */
+public interface GeneralSparkParameterComposer {
```

Comment:
why we want to differenate General and DataSource composer, they are both Function<SparkSubmitParamaters, SparkParamaters>, right?

Reply:
As discussed offline, DataSource composer is specific to Datasource related parameters, and come with DataSource Type validation (If there is no composer for the DataSource type, it would raise exception).
Added comments to make it clear.